modules/javafx.graphics/src/main/java/javafx/scene/robot/Robot.java

```diff
@@ -0,0 +1,373 @@
+     */
+    public void keyRelease(KeyCode keyCode) {
+        peer.keyRelease(keyCode);
+    }
+
+    /**
+     * Types the specified {@link KeyCode} key.
+     *
+     * @implSpec This is a convenience method that is equivalent to calling
+     * {@link #keyPress(KeyCode)} followed by {@link #keyRelease(KeyCode)}.
+     * @param keyCode the key to type
+     * @throws IllegalStateException if this method is called on a thread
+     * other than the JavaFX Application Thread.
+     */
+    public void keyType(KeyCode keyCode) {
+        keyPress(keyCode);
+        keyRelease(keyCode);
+    }
+
+    /**
+     * Returns the current mouse {@code x} position in screen coordinates.
+     *
+     * @return the current mouse {@code x} position
+     * @throws IllegalStateException if this method is called on a thread
+     * other than the JavaFX Application Thread.
+     */
+    public double getMouseX() {
+        return peer.getMouseX();
+    }
+
+    /**
+     * Returns the current mouse {@code y} position in screen coordinates.
+     *
+     * @return the current mouse {@code y} position
+     * @throws IllegalStateException if this method is called on a thread
+     * other than the JavaFX Application Thread.
+     */
+    public double getMouseY() {
+        return peer.getMouseY();
+    }
+
+    /**
+     * Returns the current mouse (x,y) screen coordinates as a {@link Point2D}.
+     *
+     * @return the current mouse (x,y) screen coordinates
+     * @throws IllegalStateException if this method is called on a thread
+     * other than the JavaFX Application Thread.
+     */
+    public Point2D getMousePosition() {
+        return new Point2D(getMouseX(), getMouseY());
+    }
+
+    /**
+     * Moves the mouse to the specified (x,y) screen coordinates.
+     *
+     * @param x screen coordinate x to move the mouse to
+     * @param y screen coordinate y to move the mouse to
+     * @throws IllegalStateException if this method is called on a thread
+     * other than the JavaFX Application Thread.
+     */
+    public void mouseMove(double x, double y) {
+        peer.mouseMove(x, y);
+    }
+
+    /**
+     * Moves the mouse to the (x,y) screen coordinates specified by the
+     * given {@code location}.
+     *
+     * @param location the (x,y) coordinates to move the mouse to
+     * @throws IllegalStateException if this method is called on a thread
+     * other than the JavaFX Application Thread.
+     */
+    public final void mouseMove(Point2D location) {
+        Objects.requireNonNull(location);
+        mouseMove(location.getX(), location.getY());
+    }
+
+    /**
+     * Presses the specified {@link MouseButton}s.
+     *
+     * @param buttons the mouse buttons to press
+     * @throws IllegalStateException if this method is called on a thread
+     * other than the JavaFX Application Thread.
+     */
+    public void mousePress(MouseButton... buttons) {
+        peer.mousePress(buttons);
```

Comment:
npe if buttons is null?

Reply:
Seems good to throw (and document) NPE here, too.